File: 7kyu/mumbling.js
/*
This time no story, no theory. The examples below show you how to write function accum:

Examples:
accum("abcd") -> "A-Bb-Ccc-Dddd"
accum("RqaEzty") -> "R-Qq-Aaa-Eeee-Zzzzz-Tttttt-Yyyyyyy"
accum("cwAt") -> "C-Ww-Aaa-Tttt"
The parameter of accum is a string which includes only letters from a..z and A..Z.
*/

function accum(s) {
	// your code
    //Split s into an array
    s = s.split('')
    //Declare variable to hold the new String
    let str = ""
    //For each element of the array
    s.forEach((element, index) => {
        //Append to str the element in uppercase
        str += element.toUpperCase()
        //And append to str the element a number of times of its index, in lower case
        str += element.toLowerCase().repeat(index)
        //If its not the last element of the array
        if(index !== s.length-1){
            //Append and '-' to separate
            str += '-'
        }
    })
    //Return the new string
    return str
}

File: python/7ky/mumbling.py
'''
This time no story, no theory. The examples below show you how to write function accum:

Examples:
accum("abcd") -> "A-Bb-Ccc-Dddd"
accum("RqaEzty") -> "R-Qq-Aaa-Eeee-Zzzzz-Tttttt-Yyyyyyy"
accum("cwAt") -> "C-Ww-Aaa-Tttt"
The parameter of accum is a string which includes only letters from a..z and A..Z.
'''

def accum(s):
    # your code
    temp = []
    for i in range(len(s)):
        if i == 0:
            temp.append(s[i].upper())
        else:
            word = ''.join([s[i].upper(),s[i].lower()*i])
            temp.append(word)
    return '-'.join(temp)

def accum(s):
    return '-'.join(c.upper() + c.lower() * i for i, c in enumerate(s))